workflow-controller/controller-service/src/main/java/io/hyscale/controller/builder/K8sAuthConfigBuilder.java

```diff
@@ -34,14 +42,33 @@
 
     @Autowired
     private ControllerConfig controllerConfig;
-
+  
     /**
      * Gets {@link K8sConfigFileAuth} from {@link ControllerConfig} default config
      * @return {@link K8sConfigFileAuth}
      */
-    public AuthConfig getAuthConfig() {
+    public AuthConfig getAuthConfig() throws HyscaleException{
+    	validate(controllerConfig.getDefaultKubeConf(), true, ControllerErrorCodes.KUBE_CONFIG_NOT_FOUND);
         K8sConfigFileAuth k8sAuth = new K8sConfigFileAuth();
         k8sAuth.setK8sConfigFile(new File(controllerConfig.getDefaultKubeConf()));
         return k8sAuth;
     }
+    
+    public AuthConfig getAuthConfig(String kubeConfigPath) throws HyscaleException{
+    	validate(kubeConfigPath, true, ControllerErrorCodes.KUBE_CONFIG_NOT_FOUND);
+        K8sConfigFileAuth k8sAuth = new K8sConfigFileAuth();
+        k8sAuth.setK8sConfigFile(new File(kubeConfigPath));
+        return k8sAuth;
+    }
+    
+    private void validate(String path, boolean kubeConf, HyscaleErrorCode hyscaleErrorCode) throws HyscaleException {
+        File conffile = new File(path);
```

Comment:
camelcase , confFile

Reply:
It is a private method, no question of having docker config json . So why boolean and error code as a parameter ?